### PR TITLE
DCS-1467 add prisoner search page for unexpected arrivals

### PR DESCRIPTION
**Pull request by GurnankCheema**
1. integration tests to be done later
2. validation to be done later
3. the /record-found, /no-records-found and /possible-records-found are just placeholder pages. Body content to be added later. 

**view**
Screenshot 2022-03-21 at 13 02 33

### CODE REVIEW

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import type { RequestHandler } from 'express'
```
Comment (GurnankCheema) [inline server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts:1]: just a basic placeholder page

File: server/views/pages/unexpectedArrivals/multipleExistingRecordsFound.njk [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{% extends "../../partials/layout.njk" %}
```
Comment (GurnankCheema) [inline server/views/pages/unexpectedArrivals/multipleExistingRecordsFound.njk:1]: just a basic placeholder page

File: server/views/pages/unexpectedArrivals/noExistingRecordsFound.njk [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{% extends "../../partials/layout.njk" %}
```
Comment (GurnankCheema) [inline server/views/pages/unexpectedArrivals/noExistingRecordsFound.njk:1]: just a basic placeholder page

File: server/views/pages/unexpectedArrivals/singleExistingRecordFound.njk [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{% extends "../../partials/layout.njk" %}
```
Comment (GurnankCheema) [inline server/views/pages/unexpectedArrivals/singleExistingRecordFound.njk:1]: just a basic placeholder page

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/validation/searchForExistingRecordsValidation.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { ValidationError, Validator } from '../../../../../middleware/validationMiddleware'
+
+const SearchForExistingRecordsValidator: Validator = (): ValidationError[] => {
+  return []
```
Comment (GurnankCheema) [inline server/routes/bookedtoday/unexpectedArrivals/arrivals/validation/searchForExistingRecordsValidation.ts:4]: validation to be done when later in sep tecket

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+import type { ExpectedArrivalsService } from '../../../../services'
+
+export default class MultipleExistingRecordsFoundController {
+  public constructor(private readonly expectedArrivalsService: ExpectedArrivalsService) {}
+
```
Comment (louisenorris) [inline server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts:6]: I think maybe best not to pass services down to controllers until we need them there

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/noExistingRecordsFoundController.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import type { RequestHandler } from 'express'
+import type { ExpectedArrivalsService } from '../../../../services'
+
+export default class NoExistingRecordsFoundController {
+  public constructor(private readonly expectedArrivalsService: ExpectedArrivalsService) {}
+
```
Comment (louisenorris) [inline server/routes/bookedtoday/unexpectedArrivals/arrivals/noExistingRecordsFoundController.ts:6]: I think maybe best not to pass services down to controllers until we need them there

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/singleExistingRecordFoundController.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import type { RequestHandler } from 'express'
+import type { ExpectedArrivalsService } from '../../../../services'
+
+export default class SingleMatchingRecordFoundController {
+  public constructor(private readonly expectedArrivalsService: ExpectedArrivalsService) {}
+
```
Comment (louisenorris) [inline server/routes/bookedtoday/unexpectedArrivals/arrivals/singleExistingRecordFoundController.ts:6]: I think maybe best not to pass services down to controllers until we need them there

File: server/routes/bookedtoday/unexpectedArrivals/index.ts [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+import Role from '../../../authentication/role'
+import validationMiddleware from '../../../middleware/validationMiddleware'
+import searchForExistingRecordsValidator from './arrivals/validation/searchForExistingRecordsValidation'
+
+export default function routes(services: Services): Router {
+  const router = express.Router()
+
+  const get = (path: string, handlers: RequestHandler[], authorisedRoles?: Role[]) =>
+    router.get(
+      path,
```
Comment (louisenorris) [inline server/routes/bookedtoday/unexpectedArrivals/index.ts:18]: maybe include the subfolders in the path parameter to reduce repetition below similar as we do in `searchforexisting/index.ts `
`/manually-confirm-arrival/search-for-existing-record${path}`,

File: server/routes/bookedtoday/unexpectedArrivals/index.ts [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+      authorisationForUrlMiddleware(authorisedRoles),
+      handlers.map(handler => asyncMiddleware(handler))
+    )
+
+  const post = (path: string, handlers: RequestHandler[], authorisedRoles?: Role[]) =>
+    router.post(
+      path,
```
Comment (louisenorris) [inline server/routes/bookedtoday/unexpectedArrivals/index.ts:25]: maybe include the subfolders in the path parameter to reduce repetition below similar as we do in searchforexisting/index.ts :
/manually-confirm-arrival/search-for-existing-record${path},

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts (hunk 6 above)
Comment (louisenorris) [reply server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts:6]: Also maybe the arrival subfolder within unexpectedArrivals is unnecessary at this stage?

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/validation/searchForExistingRecordsValidation.ts (hunk 5 above)
Comment (louisenorris) [reply server/routes/bookedtoday/unexpectedArrivals/arrivals/validation/searchForExistingRecordsValidation.ts:4]: Probably best to remove validation code while it's not doing anything and add it when validation functionality is implemented in a later PR

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts (hunk 6 above)
Comment (GurnankCheema) [reply server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts:6]: agreed

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/singleExistingRecordFoundController.ts (hunk 8 above)
Comment (GurnankCheema) [reply server/routes/bookedtoday/unexpectedArrivals/arrivals/singleExistingRecordFoundController.ts:6]: agreed

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts (hunk 6 above)
Comment (GurnankCheema) [reply server/routes/bookedtoday/unexpectedArrivals/arrivals/multipleExistingRecordsFoundController.ts:6]: agreed

File: server/routes/bookedtoday/unexpectedArrivals/index.ts (hunk 9 above)
Comment (GurnankCheema) [reply server/routes/bookedtoday/unexpectedArrivals/index.ts:18]: agreed

File: server/routes/bookedtoday/unexpectedArrivals/arrivals/validation/searchForExistingRecordsValidation.ts (hunk 5 above)
Comment (GurnankCheema) [reply server/routes/bookedtoday/unexpectedArrivals/arrivals/validation/searchForExistingRecordsValidation.ts:4]: agreed

**Pull request merged**